src/checkout/checkout-service.ts

```diff
@@ -292,7 +292,7 @@ export default class CheckoutService {
      * @throws `OrderFinalizationNotRequiredError` error if order finalization
      * is not required for the current order at the time of execution.
      */
-    finalizeOrderIfNeeded(options?: RequestOptions): Promise<CheckoutSelectors> {
+    finalizeOrderIfNeeded(options?: PaymentFinalizeOptions): Promise<CheckoutSelectors> {
```

Comment:
This is a significant change to the public interface of the library. Order finalisation, as the name of the method suggests, can only happen after an order is created and payment details are received. Therefore, it does not make much sense for this method to accept payment details as one of its parameters. Payment details should therefore only be passed through the `submitOrder` method.

Reply:
I need to add an ability for Bolt team to pass paymentData to the `checkoutService.finalizeOrderIfNeeded` method, to pass `bolt transaction reference` and submit payment.